custom/icds_reports/static/js/directives/download/download.directive.js

```diff
@@ -194,6 +194,31 @@ function DownloadController($rootScope, $location, locationHierarchy, locationsS
         vm.selectedLocations = new Array(maxLevel);
     };
 
+    vm.userHaveAccessToAllLocations = function (locations) {
+        var haveAccessToAllLocationsForLevel = true;
+        window.angular.forEach(locations, function (location) {
+            if (!location.user_have_access) {
+                haveAccessToAllLocationsForLevel = false;
+            }
+        });
+        return haveAccessToAllLocationsForLevel;
+    };
+
+    vm.userLocationIdIsNull = function () {
+        return ["null", "undefined"].indexOf(vm.userLocationId) !== -1;
+    };
+
+    vm.isUserLocationIn = function (locations) {
+        var userLocationInSorted = _.filter(locations, function (location) {
+            return allUserLocationId.indexOf(location.location_id) !== -1;
+        });
+        return userLocationInSorted.length > 0;
+    };
+
+    vm.preventShowingAllOption = function (locations) {
+        return ((!vm.userLocationIdIsNull() && !vm.userHaveAccessToAllLocations(locations)) || vm.isUserLocationIn(locations)) && !haveAccessToAllLocations;
```

Comment:
Long conditionals are difficult to read and easy to get wrong. Can you extract sub-expressions as named variables?